src/resources.rs

```diff
@@ -4,16 +4,45 @@ use std::collections::hash_map::{Entry, HashMap};
 use std::hash::{BuildHasherDefault, Hasher};
 use std::ops::{Deref, DerefMut};
 
+/// A type map for holding resources.
+///
+/// In an ECS, resources are "global" singular objects that may be accessed by systems
+/// but are not realized as [Entities](struct.Entity.html), as an optimization.
+///
+/// # Examples
+///
+/// ```
+/// # use rs_ecs::*;
+/// # struct MyResource(u32);
+/// let mut resources = Resources::new();
+///
+/// // Insert multiple resources
+/// resources.insert(42_u32);
+/// resources.insert(MyResource(0));
+///
+/// // Borrow a resource of a type, immutably
+/// let my_res = resources.get::<MyResource>();
+///
+/// // Borrow a resource of a type, mutably
+/// let mut u32_res = resources.get_mut::<u32>();
+/// *u32_res += 1;
+/// ```
 #[derive(Default)]
```

Comment:
I wonder if it would actually be lines well spent to manually implement `Default` to be able to document it as you did with `World`?